src/core/primitives.rs

```diff
@@ -451,3 +438,53 @@ fn refresh_coordinate_keys(
         })
         .collect::<RevisionVec<_, _>>()
 }
+
+/// Attempts opening the Covercrypt encapsulation using the given USK. Returns
+/// the encapsulated key and associated rights upon success, otherwise returns `None`.
+pub fn full_decaps(
+    encapsulation: &XEnc,
+    msk: &MasterSecretKey,
+) -> Result<Vec<(Right, Secret<SHARED_SECRET_LENGTH>)>, Error> {
+    // A = ⊙ _i (α_i. c_i)
+    let A = encapsulation.c.first().unwrap() * &(&msk.tsk.s / &msk.tsk.tracers.front().unwrap().0);
+    let mut rights_list: Vec<(Right, Secret<SHARED_SECRET_LENGTH>)> = Vec::new();
```

Comment:
`list` brings no information, then you can initialize the vector with the correct capacity.

```suggestion
    let mut right_to_secrets = Vec::with_capacity(encapsulation.encapsulations.len());
```